hardware/Makefile

```diff
@@ -88,14 +88,22 @@ vlog_args += -suppress vlog-2583 -suppress vlog-13314 -suppress vlog-13233
 vlog_args += -work $(library)
 # Defines
 vlog_defs += -DNUM_CORES=$(num_cores) -DNUM_CORES_PER_TILE=$(num_cores_per_tile) -DNUM_GROUPS=$(num_groups) -DBANKING_FACTOR=$(banking_factor)
-vlog_defs += -DL2_BASE=$(l2_base) -DL2_SIZE=$(l2_size) -DL2_BANKS=$(l2_banks)
-vlog_defs += -DBOOT_ADDR=$(boot_addr) -DXPULPIMG=$(xpulpimg)
+vlog_defs += -DL2_BASE=32\'d$(l2_base) -DL2_SIZE=32\'d$(l2_size) -DL2_BANKS=$(l2_banks)
+vlog_defs += -DBOOT_ADDR=32\'d$(boot_addr) -DXPULPIMG=$(xpulpimg)
 vlog_defs += -DSNITCH_TRACE=$(snitch_trace)
 vlog_defs += -DAXI_DATA_WIDTH=$(axi_data_width)
 vlog_defs += -DRO_LINE_WIDTH=$(ro_line_width)
 vlog_defs += -DDMAS_PER_GROUP=$(dmas_per_group)
 vlog_defs += -DAXI_HIER_RADIX=$(axi_hier_radix) -DAXI_MASTERS_PER_GROUP=$(axi_masters_per_group)
 vlog_defs += -DSEQ_MEM_SIZE=$(seq_mem_size) -DXQUEUE_SIZE=$(xqueue_size)
+ifdef terapool
```

Comment:
We should define all defines for all systems if possible and have as few ifdefs as possible. So you can actually just remove this ifdef.

Reply:
Hi Sam, the only thing left for now should be these `ifdef` stuff, if you agree, we can keep it temporary for this time, then we can discuss how to make it better when we have time in the future.

Reply:
We can leave them in the RTL, but we could remove them in the Makefiles, right? Then, we are sure to have all the parameters in all configurations, even if some don't do anything. It is better to have them at a reasonable default than not defined for MemPool.

Reply:
Additionally, I've used `ifdef TERAPOOL` in all the places where I made changes for the TeraPool RTL. If I set `vlog_defs += -DTERAPOOL=$(terapool)` in the `Makefile` for every configuration, it still presents an issue. This is because `TERAPOOL` will always be defined, even in `MemPool/MinPool` configurations, but with an "Empty" value. 
In this case, I think use the condition statement in Makefile is most easy way for it:
```
ifdef terapool
  vlog_defs += -DTERAPOOL=$(terapool)
 ```
 Any further suggestions?

Reply:
Hey Yichao,

About the first aspect, yes, that is exactly what I had in mind as well. But instead of defining it in every configuration file, we just define a default in the `config/config.mk` file. We can also add a comment along the lines of: "This parameter is only used for TeraPool configurations". We did similar things with the systolic parameters, like the xqueue size. It's always defined in the main config file but only used in systolic configurations.

About the `TERAPOOL` define. We could add something to the makefile that always creates a define in the name of the design/config. So something like this potentially: `vlog_defs += -D$(config)` What do you think?

Reply:
Hi Sam,

Thanks so much for your suggestions. Now, I have corrected it again. Please help to have a look.
If it looks good for you, I hope we can merge it to the main branch before Xmas!
Please let me know if something still looks bad; thanks a lot for your help!

Cheers,
Yichao.